File: src/main/java/com/morak/performancetracker/aop/rest/RestAdvisor.java
package com.morak.performancetracker.aop.rest;

import org.aopalliance.aop.Advice;
import org.springframework.aop.Pointcut;
import org.springframework.aop.PointcutAdvisor;

public class RestAdvisor implements PointcutAdvisor {

    private final Pointcut pointcut;
    private final Advice advice;

    public RestAdvisor(Pointcut pointcut, Advice advice) {
        this.pointcut = pointcut;
        this.advice = advice;
    }

    @Override
    public Pointcut getPointcut() {
        return pointcut;
    }

    @Override
    public Advice getAdvice() {
        return advice;
    }

    @Override
    public boolean isPerInstance() {
        return true;
    }
}

File: src/main/kotlin/com/morak/performancetracker/aop/rest/RestAdvisor.kt
package com.morak.performancetracker.aop.rest

import org.aopalliance.aop.Advice
import org.springframework.aop.Pointcut
import org.springframework.aop.PointcutAdvisor

class RestAdvisor(private val pointcut: Pointcut, private val advice: Advice) : PointcutAdvisor {
    override fun getPointcut(): Pointcut {
        return pointcut
    }

    override fun getAdvice(): Advice {
        return advice
    }

    override fun isPerInstance(): Boolean {
        return true
    }
}
